src/main/java/servlets/SpendingAnalyticsServlet.java

```diff
@@ -33,18 +36,30 @@
 /** Servlet that computes user spending analytics from datastore data. */
 @WebServlet("/compute-analytics")
 public class SpendingAnalyticsServlet extends HttpServlet {
+  private static final String AUTHENTICATION_ERROR_MESSAGE =
+      "No Authentication: User must be logged in to search receipts.";
+
   private final DatastoreService datastore;
+  private final UserService userService;
```

Comment:
`userService` can be initialized here since both constructors create the same object.